system/lib/wasmfs/file_table.cpp

```diff
@@ -60,4 +60,30 @@ FileTable::Handle::addEntry(std::shared_ptr<OpenFileState> openFileState) {
   return -EBADF;
 }
 
+int OpenFileState::create(std::shared_ptr<File> file,
+                          oflags_t flags,
+                          std::shared_ptr<OpenFileState>& out) {
+  assert(file);
```

Comment:
I see this is moved code, but this assert seems a little excessive? Is there a specific reason to worry about a bug here (and SAFE_HEAP/ASan are not enough)?

Reply:
No specific reasons, but I do think the assertion is reasonable. The function signature alone does not precisely convey the invariant that `file` should be non-null, and I don't think this assertion adds much cognitive overhead. If anything, I'd say it's helpful as documentation.